### ISSUE TITLE
[enhancement] Add a parameter to set maximum failure message length or a parameter don't include error message

### ISSUE THREAD

**Issue by vince-duloup** [gotestyourself/gotestsum#260]
Hello gotestsum team.
I use gotestsum to generate reports for GitHub Actions.  GitHub Actions  have limitations: "The report size cannot exceed 1 MB". 

After my test, gotestsum-report.xml contains the full log of the failed test. The length of the log is ~ 10 MB, as a result, GitHub actions cannot process this file. 

**Is it possible to add a parameter?:** 
MaximumErrorMessageLength or Don't_Include_ErrorMessage.

By the way, I tried to find a way to not include the full log into gotestsum-report.xml .  It sounds more convenient to add only an assert or panic stack trace into test report, but I have not found this way.

An example of xml:
`<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
	<testsuite tests="3" failures="2" time="6.359000" name="github.com/myProject/..." timestamp="2022-06-24T11:45:24+03:00">
		<properties>
			<property name="go.version" value="go1.18beta1 darwin/arm64"></property>
		</properties>
		<testcase classname="github.com/myProject/..." name="MyTest/test1" time="2.500000">
			<failure message="Failed" type=""> Message with lenght > 1mb</failure>
		</testcase>
	</testsuite>
</testsuites>`

**Comment by dnephin** [comment]
Thanks for opening this issue! I think this is similar to #136.  I think adding an option to limit the output would be good.

I don't think there is anyway to split the assertion messages from log output from `gotestsum`, but you could do something from the test suite to send log output to a file instead of to stdout.

**Comment by vince-duloup** [comment]
I got it, thank you for your advice!
Do I understand correctly that if I send the output to file, junit.xml will not contain the error message but still contain tests statuses?

**Comment by dnephin** [comment]
Yes, exactly! `gotestsum` only sees output sent to stdout or stderr. So if you write the output to a separate file then the junit.xml will only contain the test success/pass messages.

**Comment by dnephin** [comment]
I forgot to mention that #251 has some suggestions for how to do that.

**Comment by vince-duloup** [comment]
Thank you!
